godotproject/Player.gd

```diff
@@ -135,43 +240,84 @@ func do_movement(delta):
 			swing_angular_speed *= -0.95
 			# TODO: ^^^ Add some dampening since not every collision is perfectly elastic
 	else:
-		# Set X velocity based on user input
-		# TODO: maybe do this differently? give the frog a little X momentum?
-		velocity.x = user_direction.x * speed
+		# Adjust X velocity based on user input
+		handle_normal_horizontal_movement(delta)
 		
 		# Adjust Y velocity based on gravity and jump
 		velocity.y += gravity * delta
+
+		# If we're still in the shorthop section of our jump
 		if jump_shorthop_countdown > 0:
 			jump_shorthop_countdown -= delta
-			if jump_shorthop_countdown <= 0 and not Input.is_action_pressed("jump"):
+			# If we're still shorthopping and the user has let go of jump, then abort out of
+			# the main jump by clearing its timer, and use the shorthop speed instead.
+			if jump_shorthop_countdown <= 0 and not jump_held:
```

Comment:
have we coordinated with @jDenyes to figure out how we want to handle countdown timers?

Reply:
I think we agreed that we should be using timer modules, but I didn't change over all the pre-existing code yet. I used it for those going forward. I was planning on doing it after the demo release